extensions/resteasy-server-common/deployment/src/main/java/io/quarkus/resteasy/server/common/deployment/ResteasyServerCommonProcessor.java

```diff
@@ -229,19 +229,21 @@ public void build(
 
         Map<DotName, ClassInfo> scannedResources = new HashMap<>();
         Set<DotName> pathInterfaces = new HashSet<>();
-        Set<ClassInfo> withoutDefaultCtor = new HashSet<>();
+        Map<DotName, ClassInfo> withoutDefaultCtor = new HashMap<>();
         for (AnnotationInstance annotation : allPaths) {
             if (annotation.target().kind() == AnnotationTarget.Kind.CLASS) {
                 ClassInfo clazz = annotation.target().asClass();
                 if (!Modifier.isInterface(clazz.flags())) {
-                    String className = clazz.name().toString();
-                    if (!additionalPaths.contains(annotation)) { // scanned resources only contains real JAX-RS resources
-                        scannedResources.putIfAbsent(clazz.name(), clazz);
-                    }
-                    reflectiveClass.produce(new ReflectiveClassBuildItem(true, true, className));
+                    if (!withoutDefaultCtor.containsKey(clazz.name())) {
+                        String className = clazz.name().toString();
+                        if (!additionalPaths.contains(annotation)) { // scanned resources only contains real JAX-RS resources
+                            scannedResources.putIfAbsent(clazz.name(), clazz);
+                        }
+                        reflectiveClass.produce(new ReflectiveClassBuildItem(true, true, className));
```

Comment:
Shouldn't these be outside of the `if (!withoutDefaultCtor.containsKey(clazz.name()))` check?

Reply:
If it in `withoutDefaultCtor` it's already been processed, hasn't it?

Reply:
That is certainly true, but the point of `withoutDefaultCtor` wasn't really to guard against that case.
I do understand what your checking here, it just seems odd to reuse `withoutDefaultCtor` to perform this check.

Reply:
I could introduce another Set<DotName> for that purpose.